extensions/notebook/src/book/bookTreeView.ts

```diff
@@ -112,8 +129,18 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
 		try {
 			if (this._openAsUntitled) {
 				this.openNotebookAsUntitled(resource);
-			}
-			else {
+			} else {
+				// leave alone cached trusted statuses
+				let cache = this._trustResourceCache;
+				let normalizedResource = path.normalize(resource);
+				if (!cache[normalizedResource] && this._bookTrustManager.isNotebookTrustedByDefault(normalizedResource)) {
+					let openDocumentListenerUnsubscriber = azdata.nb.onDidOpenNotebookDocument(function (document: azdata.nb.NotebookDocument) {
+						document.setTrusted(true);
```

Comment:
How much overhead would there be to just always setTrusted to true if it's trusted by default? (suggesting get rid of the cache here)

Reply:
Not sure how often this would happen, but users are allowed to "untrust" a notebook after having opened it. Remembering which we already trusted by default will stop us from trusting it again and overriding the user's last state.

Reply:
But this is only called when the document opens - and closing a notebook clears the trusted status anyways

Reply:
Looks like closing the notebook does not clear the trust. There seems to be a cache in the notebook itself:
![cached_trust](https://user-images.githubusercontent.com/52225468/75810272-25fecd80-5d3f-11ea-89f3-9e2bd4c9835f.gif)

Reply:
Ah, yeah I didn't try opening from a book. So that seems fine then. Although I'd run this by @chlafreniere / @yualan to see if that behavior makes sense.

I'd suggest adding a comment explaining why the cache is necessary though if you do keep this logic.

Reply:
The cache isn't in the notebook. We use mementos. Take a look at src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts